packages/material-ui-unstyled/src/SwitchUnstyled/SwitchUnstyled.tsx

```diff
@@ -0,0 +1,177 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import useSwitch, { UseSwitchProps } from './useSwitch';
+import classes from './switchUnstyledClasses';
+
+export interface SwitchUnstyledProps extends UseSwitchProps {
+  /**
+   * The component used for the Root slot.
+   * Either a string to use a HTML element or a component.
+   * This is equivalent to `components.Root`. If both are provided, the `component` is used.
+   */
+  component?: React.ElementType;
+  /**
+   * The components used for each slot inside the Switch.
+   * Either a string to use a HTML element or a component.
+   * @default {}
+   */
+  components?: {
+    Root?: React.ElementType;
+    Thumb?: React.ElementType;
+    Input?: React.ElementType;
+  };
+
+  /**
+   * The props used for each slot inside the Slider.
+   * @default {}
+   */
+  componentsProps?: {
+    root?: {};
+    thumb?: {};
+    input?: {};
+  };
+}
+
+/**
+ *
+ * Demos:
+ *
+ * - [Switches](https://material-ui.com/components/switches/)
+ *
+ * API:
+ *
+ * - [SwitchUnstyled API](https://material-ui.com/api/switch-unstyled/)
+ */
+const SwitchUnstyled = React.forwardRef(function SwitchUnstyled(
+  props: SwitchUnstyledProps,
+  ref: React.ForwardedRef<any>,
+) {
+  const {
+    component,
+    components = {},
+    componentsProps = {},
+    onChange,
+    onBlur,
+    onFocus,
+    onFocusVisible,
+    checked: checkedProp,
+    defaultChecked,
+    disabled: disabledProp,
+    required,
+    readOnly,
+    ...otherProps
+  } = props;
+
+  const Root: React.ElementType = components.Root ?? component ?? 'span';
+  const rootProps: any = { ...otherProps, ...componentsProps.root };
+
+  const Thumb: React.ElementType = components.Thumb ?? 'span';
+  const thumbProps: any = componentsProps.thumb ?? {};
+
+  const Input: React.ElementType = components.Input ?? 'input';
+  const inputProps: any = componentsProps.input ?? {};
+
+  const useSwitchProps = {
+    onChange,
+    onBlur,
+    onFocus,
+    onFocusVisible,
+    checked: checkedProp,
+    defaultChecked,
+    disabled: disabledProp,
+  };
+
+  const { getInputProps, checked, disabled, focusVisible } = useSwitch(useSwitchProps);
+
+  const stateClasses = {
+    [classes.checked]: checked,
+    [classes.disabled]: disabled,
+    [classes.focusVisible]: focusVisible,
+  };
+
+  return (
+    <Root
+      ref={ref}
+      {...rootProps}
+      className={clsx(classes.root, stateClasses, rootProps?.className)}
```

Comment:
So, no support for the `classes` prop?

Reply:
Nope. We discussed it with @mnajdova. We already have `componentsProps` extension point and I feel that creating yet another one (that could potentially clash with `componentsProps.x.className`) won't help.

Reply:
Ok, Interesting. Let's see how this goes.